updates/commands.go

```diff
@@ -119,6 +123,111 @@ func createStatusMessage(user users.User, chatId int64, msg tgbotapi.MessageConf
 }
 
+func handleMonthlySummaryCommand(fatBotUpdate FatBotUpdate) {
+	update := fatBotUpdate.Update
+	bot := fatBotUpdate.Bot
+	originalChatId := update.FromChat().ID
+	user, err := users.GetUserFromMessage(update.Message)
+	if err != nil {
+		log.Error(err)
+		sentry.CaptureException(err)
+		return
+	}
+
+	if user.ID == 0 {
+		return
+	}
+	months := 6 // until we'll let the user enter the number of months
+	start, end := getDuration(months)
+	chatIds, err := user.GetChatIds()
+	if err != nil {
+		log.Error(err)
+		sentry.CaptureException(err)
+		return
+	}
+	if len(chatIds) > 0 {
+		for _, chatId := range chatIds {
+			createUserMonthlySummaryChart(bot, originalChatId, user, chatId, start, end)
+		}
+	} else {
+		log.Errorf("No chat IDs found for user %s", user.Username)
+	}
+}
+func createUserMonthlySummaryChart(bot *tgbotapi.BotAPI, originalChatId int64, user users.User, chatId int64, start, end time.Time) {
+	group, _ := users.GetGroup(chatId)
+	fileName := fmt.Sprintf("%s_%s.png", user.Username, group.Title)
+	months := getMonthsInRange(start, end)
+	userWorkoutsByMonthAsString := user.GetMonthlyWorkoutsSummary(group, start, end, months)
+	monthsStringSlice := "'" + strings.Join(months, "', '") + "'"
+	userWorkoutsByMonthStringSlice := strings.Join(userWorkoutsByMonthAsString, ", ")
+	chartConfig := createMonthlyChartConfig(monthsStringSlice, userWorkoutsByMonthStringSlice)
+	qc := createQuickChart(chartConfig)
+	file, err := os.Create(fileName)
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+	qc.Write(file)
+	msg := tgbotapi.NewPhoto(originalChatId, tgbotapi.FilePath(fileName))
+	msg.Caption = fmt.Sprintf("Monthly Workouts summary\n")
+
+	if _, err = bot.Send(msg); err != nil {
+		log.Error(err)
+		sentry.CaptureException(err)
+	}
+}
+
+func createMonthlyChartConfig(monthsStringSlice, userWorkoutsByMonthStringSlice string) string {
+	chartConfig := fmt.Sprintf(`{
+		type: 'bar',
+		data: {
+			labels: [%s],
+			datasets: [
+				{
+					label: 'Workouts',
+					data: [%s]
+				},
+				]
+			}
+			}`, monthsStringSlice, userWorkoutsByMonthStringSlice)
+	return chartConfig
+}
+
+func getDuration(months int) (firstDayOfStartMonth, currentTime time.Time) {
+	currentTime = time.Now()
+	firstDayOfCurrentMonth := time.Date(currentTime.Year(), currentTime.Month(), 1, 0, 0, 0, 0, currentTime.Location())
+	firstDayOfStartMonth = firstDayOfCurrentMonth.AddDate(0, -months, 0)
+	return
+
+}
+
+func getMonthsInRange(start, end time.Time) []string {
+	months := make([]string, 0)
+	currentMonth := start.Month()
+	currentYear := start.Year()
+	for start.Before(end) || start.Equal(end) {
+		months = append(months, start.Month().String())
+		currentMonth++
+		if currentMonth > 12 {
+			currentMonth = 1
+			currentYear++
+		}
+
+		// Set the start to the first day of the next month
+		start = time.Date(currentYear, currentMonth, 1, 0, 0, 0, 0, start.Location())
+	}
+	return months
+}
+
+func createQuickChart(chartConfig string) *quickchartgo.Chart {
```

Comment:
The exact same function and config exist in `reports.go`.
Which brings me to three topics here  - 
1. Maybe move the entire "report handling" code to `reports.go`
2. DRY - Don't repeat yourself 😉  use the same
3. Maybe refactor both to Viper config and grab from there? 🤷